dashboard-api/internal/platform/server/handler/studies/list.go

```diff
@@ -0,0 +1,113 @@
+package studies
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	studiesmanager "github.com/vlab-research/vlab/dashboard-api/internal"
+)
+
+type studyReponse struct {
+	Id        string `json:"id"`
+	Name      string `json:"name"`
+	Slug      string `json:"slug"`
+	CreatedAt int64  `json:"createdAt"`
+}
+
+func ListHandler(studyRepository studiesmanager.StudyRepository) gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		pagination, err := parsePaginationQueryParams(ctx)
+		if err != nil {
+			ctx.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+			return
+		}
+
+		studies, err := studyRepository.GetStudies(ctx, pagination.Cursor, pagination.Number)
+		if err != nil {
+			ctx.Status(http.StatusInternalServerError)
+			return
+		}
+
+		data := make([]studyReponse, 0, len(studies))
```

Comment:
Not actually a deal because the performance difference is negligible, but I find it confusing to specify the capacity and then use append. Why not just specify the length and assign?

Reply:
I got rid of this code, since I'm returning the entity directly it's not necessary anymore.